### ISSUE TITLE
Possible to use WDC as a node module in a React.js app?

### ISSUE THREAD

**Issue by tehandyb** [tableau/webdataconnector#137]
Hi I see that you can include the WDC globally, but was wondering if I could install it as a node module for use in a React.js app. I also was wondering if there was any information on how to go about using the WDC in a React.js app.

**Comment by lbrendanl** [comment]
Hey @tehandyb, we do not have the WDC API distributed as a node package.  You should just be able to include it as a regular script in your React app; not as elegant but it'll work.  

We don't have any explicit guidance about how to use WDC within the react framework, but it shouldn't be any different than using it in a plain html/javascript page.

**Comment by tehandyb** [comment]
Ok thanks, I'll include it globally like that and try it out, thanks!

**Comment by tehandyb** [comment]
It did work. But now I'm running into another issue with the WDC in my react app. Currently my app makes an asyncronous call for some data, but it does this in a flux store, so it's behind the scenes. Therefore I am not able to give the connector.getData function an async function to call. Instead I would like to register my connector once I get the data in the application, but it seems I get an error if I wait to register the connector and then the rest of the connector doesn't work anymore. 

TLDR:
Is there a way to register my connector later on? Or register it again but with a new connector.getData() function that holds the new data that I want? I tried registering the connector again but it seems that the old getData function gets executed instead of the new one.

**Comment by tehandyb** [comment]
Going to make another issue, since the new one is separate from my original issue.

**Comment by PurpleUltralisk** [comment]
Hi tehandyb, 
I'm new to development, just wondering if you can help explain how you included the WDC globally in your node.js file? 
Thanks!

**Comment by tehandyb** [comment]
@PurpleUltralisk Sure, what I mean by globally - is including the required Tableau Javascript file in my html. So adding something like the following somewhere between your html body tags(preferably just before the closing body tag): 
```
<script src="https://connectors.tableau.com/libs/tableauwdc-2.0.latest.js"></script>
```
Then the Tableau functions will be available to any javascript that you add after that.
You can find the different library versions and their compatibility with the different Tableau versions here http://tableau.github.io/webdataconnector/docs/wdc_library_versions

**Comment by duongthienlee** [comment]
Do you mind sharing your repo ?
I'm struggling to make web data connector work with React.
